rct229/rules/section5/section5rule37.py

```diff
@@ -0,0 +1,258 @@
+                    SCC.EXTERIOR_NON_RESIDENTIAL,
+                    OST.SKYLIGHT,
+                    skylit_wwr=0.02,
+                )
+                if srr_nonres > 0.02
+                else table_G34_lookup(
+                    climate_zone,
+                    SCC.EXTERIOR_NON_RESIDENTIAL,
+                    OST.SKYLIGHT,
+                    skylit_wwr=0.03,
+                )
+            )
+            # Process target_u_factor_semiheated
+            srr_semi_exterior = building_scc_skylight_roof_ratios_dict_b[
+                SCC.SEMI_EXTERIOR
+            ]
+            target_u_factor_semiheated = (
+                table_G34_lookup(
+                    climate_zone,
+                    SCC.SEMI_EXTERIOR,
+                    OST.SKYLIGHT,
+                    skylit_wwr=0.02,
+                )
+                if srr_semi_exterior > 0.02
+                else table_G34_lookup(
+                    climate_zone,
+                    SCC.SEMI_EXTERIOR,
+                    OST.SKYLIGHT,
+                    skylit_wwr=0.03,
+                )
+            )
+
+            return {
+                **data,
+                "surface_conditioning_category_dict_b": get_surface_conditioning_category_dict(
+                    climate_zone, building_b
+                ),
+                # at this point, target_u_factor_mixed should be same regardless of
+                # residential <2% or >2%, skylight.
+                "target_u_factor_res_b": target_u_factor_res["u_value"],
+                "target_u_factor_nonres_b": target_u_factor_nonres["u_value"],
+                "target_u_factor_semiheated_b": target_u_factor_semiheated["u_value"],
+            }
+
+        def list_filter(self, context_item, data=None):
+            # context_item shall be the list of the list_path element
+            surface_b = context_item.baseline
+            # roof with subsurfaces, and the roof is not unregulated
+            return (
+                get_opaque_surface_type(surface_b) == OST.ROOF
+                and surface_b.get("subsurfaces", None)
+                and data["surface_conditioning_category_dict_b"][surface_b["id"]]
+                != SCC.UNREGULATED
+            )
+
+        class RoofRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section5Rule37.BuildingRule.RoofRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    each_rule=Section5Rule37.BuildingRule.RoofRule.SubsurfaceRule(),
+                    index_rmr="baseline",
+                    list_path="subsurfaces[*]",
+                )
+
+            def create_data(self, context, data=None):
+                surface_b = context.baseline
+                scc_type = data["surface_conditioning_category_dict_b"][surface_b["id"]]
+                return {**data, "scc_type": scc_type}
+
+            class SubsurfaceRule(RuleDefinitionBase):
```

Comment:
As I read the Rule Assertion - RMR section at the bottom of the RDS, the RDS seems to want an output for each roof, not each subsurface of the roof.  So I think all the logic needs to take place inside RoofRule. In other words, I don't think there should be a SubsurfaceRule here.

Reply:
The [component level](https://github.com/pnnl/ruleset-checking-tool/blob/develop/docs/section5/Rule5-37.md?plain=1#L71-L81) rule assertion seems mixed surface and subsurfaces.
I think we still need to check each subsurface to verify the U-factor matches the table.

At the RMR level, it also seems we need to count the number of [subsurfaces ](https://github.com/pnnl/ruleset-checking-tool/blob/develop/docs/section5/Rule5-37.md?plain=1#L85-L89) for outcome reporting. My understanding is each subsurface has a rule check so we can count the number of fail, undetermined or success